src/engine/srv.c

```diff
@@ -335,6 +336,17 @@ wait_all_exited(struct dss_xstream *dx)
 		if (total_size == 0)
 			break;
 
+		/*
+		 * Abort timeout rpc properly, otherwise some ult like(DTX)
+		 * will wait forever.
+		 */
+		if (dx->dx_comm) {
+			rc = crt_progress(dmi->dmi_ctx, 0);
```

Comment:
We might need issue a barrier to stop accepting RPC requests before this.

Reply:
Why not export crt_context_timeout_check() and directly call it here?